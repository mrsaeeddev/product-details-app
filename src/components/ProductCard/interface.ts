import { product } from "../ProductGrid/interface";

export interface productProps {
    product: product
}