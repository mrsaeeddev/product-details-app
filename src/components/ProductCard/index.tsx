import React from "react";
import { View, Text, Image, StyleSheet } from "react-native";
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { BLUE_COLOR, FONT_WEIGHT, PADDING, PADDING_HORIZONTAL, PADDING_VERTICAL, PRIMARY_COLOR } from "../../constants/styling";
import { globalStyles } from "../../globalStyles";
import { productProps } from "./interface";

const styles = StyleSheet.create({
    coverImage: {
        width: wp('42%'),
        padding: PADDING,
        height: 200,
        borderRadius: 20,
    },
    item: {
        width: wp('50%'),
        alignItems: "center",
        paddingLeft: PADDING_HORIZONTAL,
        paddingTop: PADDING_VERTICAL,
        paddingRight: PADDING_HORIZONTAL,
        justifyContent: "flex-start"
    },
    content: {
        width: wp('42%'),
        alignItems: "flex-start",
    },
    brandName: {
        color: BLUE_COLOR,
        paddingTop: 5,
        paddingBottom: 5,
    }
});

function ProductCard(props: productProps) {
    return (<View style={styles.item}><Image
        style={styles.coverImage}
        source={require('../../images/personalcare.jpg')}
    />
        <View style={styles.content}>
            <Text style={[globalStyles.boldText, styles.brandName]}>{props.product.brand.name}</Text>
            <Text style={globalStyles.boldText}>{props.product.name}</Text>
            <Text style={globalStyles.boldText}>{props.product.price}</Text>
        </View>
    </View>)
}

export default ProductCard;
