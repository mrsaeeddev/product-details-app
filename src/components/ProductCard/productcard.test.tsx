import React from 'react';
import renderer from 'react-test-renderer';

import ProductCard from './';

describe('ProductCard', () => {
    it('should render correctly using Snapshot', () => {
        const product = {
            name: 'dummy',
            brand: { name: 'dummy' },
            price: 12,
            img: 'dummy_url',
        }
        const tree = renderer.create(<ProductCard product={product} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});