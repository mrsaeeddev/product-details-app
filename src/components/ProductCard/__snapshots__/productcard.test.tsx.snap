// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render correctly using Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "flex-start",
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 15,
      "width": 375,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "dummy_url",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 200,
        "padding": 20,
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "width": 315,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          Object {
            "color": "#83a2be",
            "paddingBottom": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      dummy
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      dummy
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      12
    </Text>
  </View>
</View>
`;
